### ISSUE TITLE
Calling `set_brightness` inside a function bound to `pynput` library's `mouse.Listener` results in error

### ISSUE THREAD

**Issue by anay-p** [Crozzers/screen_brightness_control#42]
**Description**
You can run the code below to reproduce the error. I've just declared a function `on_scroll` that calls `sbc.set_brightness` within it. `on_scroll` is bound to `pynput.mouse.Listener`. The `on_click` function is added to be able to exit out of the program and it is bound to the listener as well. When you run this code in a python environment with `pynput` installed and try scrolling up or down, you will encounter this error. Press any mouse button to exit out of the program.

**Code to reproduce**
```python
from pynput import mouse
import screen_brightness_control as sbc

def on_click(*_):
    return False

def on_scroll(x, y, dx, dy):
    if dy == 1:
        sbc.set_brightness('+5')
    elif dy == -1:
        sbc.set_brightness('-5')

with mouse.Listener(on_click=on_click, on_scroll=on_scroll) as listener:
    listener.join()
```

**Error**
```
Unhandled exception in listener callback
Traceback (most recent call last):
  File "C:\Python\Python312\Lib\site-packages\pynput\_util\win32.py", line 386, in _handler
    converted = self._convert(code, msg, lpdata)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python\Python312\Lib\site-packages\pynput\_util\win32.py", line 401, in _convert
    raise NotImplementedError()
NotImplementedError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python\Python312\Lib\site-packages\wmi.py", line 1340, in connect
    obj = GetObject(moniker)
          ^^^^^^^^^^^^^^^^^^
  File "C:\Python\Python312\Lib\site-packages\win32com\client\__init__.py", line 84, in GetObject
    return Moniker(Pathname, clsctx)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python\Python312\Lib\site-packages\win32com\client\__init__.py", line 101, in Moniker
    moniker, i, bindCtx = pythoncom.MkParseDisplayName(Pathname)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
pywintypes.com_error: (-2147417843, 'An outgoing call cannot be made since the application is dispatching an input-synchronous call.', None, None)   

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python\Python312\Lib\site-packages\pynput\_util\__init__.py", line 229, in inner
    return f(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python\Python312\Lib\site-packages\pynput\_util\win32.py", line 390, in _handler
    self._handle(code, msg, lpdata)
  File "C:\Python\Python312\Lib\site-packages\pynput\mouse\_win32.py", line 221, in _handle
    self.on_scroll(data.pt.x, data.pt.y, dd * mx, dd * my)
  File "C:\Python\Python312\Lib\site-packages\pynput\_util\__init__.py", line 145, in inner
    if f(*args) is False:
       ^^^^^^^^
  File "path\to\file\test.py", line 9, in on_scroll
    sbc.set_brightness('+5')
  File "C:\Python\Python312\Lib\site-packages\screen_brightness_control\config.py", line 16, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python\Python312\Lib\site-packages\screen_brightness_control\__init__.py", line 119, in set_brightness
    for monitor in filter_monitors(display=display, method=method, allow_duplicates=allow_duplicates):
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python\Python312\Lib\site-packages\screen_brightness_control\config.py", line 16, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python\Python312\Lib\site-packages\screen_brightness_control\__init__.py", line 672, in filter_monitors
    duplicates = get_monitor_list()
                 ^^^^^^^^^^^^^^^^^^
  File "C:\Python\Python312\Lib\site-packages\screen_brightness_control\__init__.py", line 612, in get_monitor_list
    monitors_with_duplicates = list_monitors_info(
                               ^^^^^^^^^^^^^^^^^^^
  File "C:\Python\Python312\Lib\site-packages\screen_brightness_control\config.py", line 16, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python\Python312\Lib\site-packages\screen_brightness_control\__init__.py", line 272, in list_monitors_info
    return _OS_MODULE.list_monitors_info(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python\Python312\Lib\site-packages\screen_brightness_control\windows.py", line 365, in list_monitors_info
    info = get_display_info()
           ^^^^^^^^^^^^^^^^^^
  File "C:\Python\Python312\Lib\site-packages\screen_brightness_control\windows.py", line 73, in get_display_info
    wmi = _wmi_init()
          ^^^^^^^^^^^
  File "C:\Python\Python312\Lib\site-packages\screen_brightness_control\windows.py", line 28, in _wmi_init
    return wmi.WMI(namespace='wmi')
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python\Python312\Lib\site-packages\wmi.py", line 1354, in connect
    handle_com_error()
  File "C:\Python\Python312\Lib\site-packages\wmi.py", line 258, in handle_com_error
    raise klass(com_error=err)
wmi.x_wmi: <x_wmi: Unexpected COM Error (-2147417843, 'An outgoing call cannot be made since the application is dispatching an input-synchronous call.', None, None)>
```

**Debug info**
```
================DEBUG INFO================
{'all_monitors': [{'get_brightness': [83],
                   'global_index': 0,
                   'info': {'edid': None,
                            'index': 0,
                            'manufacturer': 'Lenovo',
                            'manufacturer_id': 'LEN',
                            'method': <class 'screen_brightness_control.windows.WMI'>,
                            'model': '9059',
                            'name': 'Lenovo 9059',
                            'serial': '5&9ea0e81&0&UID256',
                            'uid': '256'},
                   'set_brightness': None}],
 'file': 'C:\\Python\\Python312\\Lib\\site-packages\\screen_brightness_control\\__init__.py',
 'filtered_monitors': [{'edid': None,
                        'index': 0,
                        'manufacturer': 'Lenovo',
                        'manufacturer_id': 'LEN',
                        'method': <class 'screen_brightness_control.windows.WMI'>,
                        'model': '9059',
                        'name': 'Lenovo 9059',
                        'serial': '5&9ea0e81&0&UID256',
                        'uid': '256'}],
 'methods': [{'class': "<class 'screen_brightness_control.windows.WMI'>",
              'displays': [{'global_index': 0, 'index': 0}],
              'name': 'wmi'},
             {'class': "<class 'screen_brightness_control.windows.VCP'>",
              'displays': [],
              'name': 'vcp'}],
 'platform': 'Windows',
 'version': '0.24.0',
 'windows': {'physical_monitor_handles': [],
             'win32api_enum_display_device_ids': ['\\\\?\\DISPLAY#LEN9059#5&9ea0e81&0&UID256#{e6f07b5f-ee97-4a90-b076-33f57bf4eaa7}'],
             'wmi_monitor_id_output': [<_wmi_object: b'\\\\LAPTOPZERO\\root\\wmi:WmiMonitorID.InstanceName="DISPLAY\\\\LEN9059\\\\5&9ea0e81&0&UID256_0"'>]}}
===================LOGS===================
DEBUG:screen_brightness_control._debug.info:gathering list of all monitors
DEBUG:screen_brightness_control.helpers.__Cache_1204459830016:'windows_monitors_info_raw' not present in cache
WARNING:screen_brightness_control.windows:exception parsing edid str for DISPLAY\LEN9059\5&9ea0e81&0&UID256_0 - EDIDParseError: parsed EDID returned invalid display name
DEBUG:screen_brightness_control.helpers.__Cache_1204459830016:cache set 'windows_monitors_info_raw', expires=1
DEBUG:screen_brightness_control._debug.info:filtering the list of monitors
DEBUG:screen_brightness_control._debug.info:get_brightness for monitor WMI:0
DEBUG:screen_brightness_control._debug.info:set_brightness for monitor WMI:0
DEBUG:screen_brightness_control._debug.info:getting display info for method: wmi
DEBUG:screen_brightness_control._debug.info:getting display info for method: vcp
DEBUG:screen_brightness_control._debug.info:windows specific debug info
```

I think this is an issue of making WMI calls in a different thread than the main one, something like that, but I'm not sure.

**Comment by Crozzers** [comment]
This appears to be an issue in the way the COM libraries are initialized, and how it handles multiple libraries using them at the same time.

It looks like there are a couple of modes used in COM which dictate how it handles threading: STA (single thread apartment) and MTA (multi thread apartment). From what I understand, STA is used for non-thread-safe objects and is commonly used in GUI components. MTA is used for thread-safe objects and can be accessed from any thread at any time.

[STA is the default](https://timgolden.me.uk/pywin32-docs/pythoncom__CoInitialize_meth.html) in the `pythoncom` package.

I tested this code using MTA and that fixed the error, although I'm hesitant to change the default behaviour as I'm not sure what the ramifications are.

What I've done is I've added a variable that can be used to configure the flags passed to the COM initialisation. In your case, you'd do something like this:
```python
import pythoncom
import screen_brightness_control as sbc
sbc.windows.COM_MODEL = pythoncom.COINIT_MULTITHREADED
...
```
The full list of available flags are available on [MS docs](https://learn.microsoft.com/en-us/windows/win32/api/objbase/ne-objbase-coinit).

I've pushed these tweaks to the [com-init-tweaks](https://github.com/Crozzers/screen_brightness_control/tree/com-init-flags) branch. Give them a try and let me know if that fixes things. If so, I'll publish a release
